### ISSUE TITLE
supervised_finetuning.py的preprocess_function函数是否有问题

### ISSUE THREAD

**Issue by baibaiw5** [shibing624/MedicalGPT#13]
### Describe the Question
使用llama tokenizer测试了一下：
tokenizer(["你好啊"], truncation=True, max_length=20)
{'input_ids': [[1, 32827, 31076, 75158]], 'attention_mask': [[1, 1, 1, 1]]}

tokenizer(["你好啊"], truncation=True, max_length=20,padding="max_length")
{'input_ids': [[1, 32827, 31076, 75158, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], 'attention_mask': [[1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]}

以下代码中tokenized_sources的长度可能<=max_source_length，没有设置：padding="max_length"
tokenized_sources = tokenizer(sources, truncation=True, max_length=max_source_length)

labels这样设置是否错误？
labels = torch.LongTensor([IGNORE_INDEX] * (max_source_length + max_target_length - len(t)) + t)

![image](https://github.com/shibing624/MedicalGPT/assets/233871/fe6125d5-1842-4fb5-8675-8bd40b00305c)

**Comment by shibing624** [comment]
没问题，是这样设计的，适配的seq2seqcollator，collator会把input_ids padding到最大长度，这样取train_dataloader时的数据是匹配的。

**Comment by baibaiw5** [comment]
> 没问题，是这样设计的，适配的seq2seqcollator，collator会把input_ids padding到最大长度，这样取train_dataloader时的数据是匹配的。

449行已经执行了preprocess,这时collator还未起作用(dataset的处理是即时处理，应该不是Lazy)：
train_dataset = train_dataset.shuffle().map(
                preprocess_function,

496行才设置：data_collator

我打印了tokenized_sources里面的内容，长度不是默认的256
